pkg/codec.go

```diff
@@ -229,8 +233,8 @@ func (m *DubboCodec) messageData(message remote.Message, e iface.Encoder) error
 		return fmt.Errorf("invalid data: not hessian2.MessageWriter")
 	}
 
-	typeAnno := m.getTypeAnnotation(message)
-	types, err := m.methodCache.GetTypes(data, typeAnno)
+	methodAnno := m.getMethodAnnotation(message)
```

Comment:
`getMethodAnnotation` is already called once in `encodeRequestPayload`, we can pass the return value into `messageData` to avoid the second call.